src/main/java/com/nu/art/cyborg/modules/InternetConnectivityModule.java

```diff
@@ -55,12 +61,7 @@ public void setTimeout(int timeout) {
 	protected void init() {
 		handler = getModule(ThreadsModule.class).getDefaultHandler("Internet Check");
 		registerReceiver(ConnectivityCheckReceiver.class);
-		postOnUI(new Runnable() {
```

Comment:
@TacB0sS was there a reason for posting this on UI ?

Reply:
i feel like this can be even more optimal.. but yeah.. no particular reason to run it on the ui thread